src/cli/one_helper/onebackupjob_helper.rb

```diff
@@ -0,0 +1,286 @@
+            end
+        end
+    end
+
+    def self.create_backupjob_template(options)
+        template_options = TEMPLATE_OPTIONS.map do |o|
+            o[:name].to_sym
+        end
+
+        template_options << :name
+
+        t = ''
+        template_options.each do |n|
+            t << "#{n.to_s.upcase}=\"#{options[n]}\"\n" if options[n]
+        end
+
+        t
+    end
+
+    private
+
+    def factory(id = nil)
+        if id
+            OpenNebula::BackupJob.new_with_id(id, @client)
+        else
+            xml=OpenNebula::BackupJob.build_xml
+            OpenNebula::BackupJob.new(xml, @client)
+        end
+    end
+
+    def factory_pool(user_flag = -2)
+        OpenNebula::BackupJobPool.new(@client, user_flag)
+    end
+
+    def format_resource(bj, options = {})
+        bj_hash = bj.to_hash
+
+        str='%-15s: %-20s'
+        str_h1='%-80s'
+
+        # ----------------------------------------------------------------------
+        CLIHelper.print_header(
+            str_h1 % "BACKUP JOB #{bj['ID']} INFORMATION"
+        )
+        # ----------------------------------------------------------------------
+        puts format(str, 'ID', bj.id.to_s)
+        puts format(str, 'NAME', bj.name)
+        puts format(str, 'USER', bj['UNAME'])
+        puts format(str, 'GROUP', bj['GNAME'])
+        puts format(str, 'LOCK', OpenNebulaHelper.level_lock_to_str(bj['LOCK/LOCKED']))
+
+        CLIHelper.print_header(str_h1 % 'PERMISSIONS', false)
+
+        ['OWNER', 'GROUP', 'OTHER'].each do |e|
+            mask = '---'
+            mask[0] = 'u' if bj["PERMISSIONS/#{e}_U"] == '1'
+            mask[1] = 'm' if bj["PERMISSIONS/#{e}_M"] == '1'
+            mask[2] = 'a' if bj["PERMISSIONS/#{e}_A"] == '1'
+
+            puts format(str, e, mask)
+        end
+        puts
+
+        # ----------------------------------------------------------------------
+        CLIHelper.print_header(
+            str_h1 % 'LAST BACKUP JOB EXECUTION INFORMATION'
+        )
+        # ----------------------------------------------------------------------
+        puts format(str, 'TIME', OpenNebulaHelper.time_to_str(bj['LAST_BACKUP_TIME']))
+        puts format(str, 'DURATION', OpenNebulaHelper.period_to_str(bj['LAST_BACKUP_DURATION']))
+        puts
+
+        # ----------------------------------------------------------------------
+        CLIHelper.print_header(
+            str_h1 % 'VIRTUAL MACHINE BACKUP STATUS'
+        )
+        # ----------------------------------------------------------------------
+        up  = bj.retrieve_elements('UPDATED_VMS/ID')
+        out = bj.retrieve_elements('OUTDATED_VMS/ID')
+        act = bj.retrieve_elements('BACKING_UP_VMS/ID')
+        err = bj.retrieve_elements('ERROR_VMS/ID')
+
+        up  = [] if up.nil?
+        out = [] if out.nil?
+        act = [] if act.nil?
+        err = [] if err.nil?
+
+        puts format(str, 'UPDATED', up.join(','))
+        puts format(str, 'OUTDATED', out.join(','))
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Potential log injection detected. Ensure all untrusted input is properly sanitized before logging. Use parameterized logging or validate input against an allow list to prevent log injection vulnerabilities. Consider using a dedicated logging library&#39;s built-in sanitization features when available. Learn more - https://cwe.mitre.org/data/definitions/117.html